core/transaction-pool/src/tests.rs

```diff
@@ -181,3 +189,29 @@ fn should_ban_invalid_transactions() {
 	// then
 	block_on(pool.submit_one(&BlockId::number(0), uxt.clone())).unwrap_err();
 }
+
+#[test]
+fn should_correctly_prune_transactions_providing_more_than_one_tag() {
+	let mut api = TestApi::default();
+	api.modifier = Box::new(|v: &mut ValidTransaction| {
+		v.provides.push(vec![155]);
+	});
+	let pool = Pool::new(Default::default(), api);
+	let xt = uxt(Alice, 209);
+	block_on(pool.submit_one(&BlockId::number(0), xt.clone())).unwrap();
+	assert_eq!(pool.status().ready, 1);
+
+	// remove the transaction that just got imported.
+	block_on(pool.prune_tags(&BlockId::number(1), vec![vec![209]], vec![])).unwrap();
+	assert_eq!(pool.status().ready, 0);
+
+	// so now let's insert another transaction that also provides the 155
+	let xt = uxt(Alice, 210);
+	block_on(pool.submit_one(&BlockId::number(1), xt.clone())).unwrap();
```

Comment:
I'd always prefer expect vs unwrap everywhere since it saves some time later when modifying logic to see what fails